File: spring-cloud-config/config-local-client/src/main/java/com/livk/config/ConfigLocalClient.java
package com.livk.config;

import com.livk.commons.spring.LivkSpring;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.http.HttpEntity;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * <p>
 * ConfigLocalClient
 * </p>
 *
 * @author livk
 * @date 2022/7/27
 */
@RestController
@SpringBootApplication
public class ConfigLocalClient {

    @Value("${app.name}")
    String name;

    public static void main(String[] args) {
        LivkSpring.run(ConfigLocalClient.class, args);
    }

    @GetMapping
    public HttpEntity<String> get() {
        return ResponseEntity.ok(name);
    }
}

File: spring-cloud-config/config-local-client/src/main/kotlin/com/livk/config/ConfigLocalClient.kt
package com.livk.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.http.HttpEntity
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

/**
 * @author livk
 */
@RestController
@SpringBootApplication
open class ConfigLocalClient {
    @Value("\${app.name}")
    lateinit var name: String


    @GetMapping("get")
    fun get(): HttpEntity<String> = ResponseEntity.ok(name)
}

fun main(args: Array<String>) {
    runApplication<ConfigLocalClient>(*args)
}
